lib/matplotlib/font_manager.py

```diff
@@ -384,10 +421,45 @@ def ttfFontProperty(font):
     else:
         variant = 'normal'
 
-    if font.style_flags & ft2font.BOLD:
-        weight = 700
-    else:
-        weight = next((w for w in weight_dict if w in sfnt4), 400)
+    # The weight-guessing algorithm is directly translated from fontconfig
+    # 2.13.1's FcFreeTypeQueryFaceInternal (fcfreetype.c).
+    wws_subfamily = 22
+    typographic_subfamily = 16
+    font_subfamily = 2
+    styles = [
+        sfnt.get((*mac_key, wws_subfamily), b'').decode('latin-1'),
+        sfnt.get((*mac_key, typographic_subfamily), b'').decode('latin-1'),
+        sfnt.get((*mac_key, font_subfamily), b'').decode('latin-1'),
+        sfnt.get((*ms_key, wws_subfamily), b'').decode('utf-16-be'),
+        sfnt.get((*ms_key, typographic_subfamily), b'').decode('utf-16-be'),
+        sfnt.get((*ms_key, font_subfamily), b'').decode('utf-16-be'),
+    ]
+    styles = [*filter(None, styles)] or [font.style_name]
+
+    def get_weight():  # From fontconfig's FcFreeTypeQueryFaceInternal.
+        # OS/2 table weight.
+        os2 = font.get_sfnt_table("OS/2")
+        if os2 and os2["version"] != 0xffff:
+            return os2["usWeightClass"]
+        # PostScript font info weight.
+        try:
+            ps_font_info = font.get_ps_font_info()
+        except ValueError:
+            pass
+        else:
+            for regex, weight in _weight_regexes:
+                if re.fullmatch(regex, ps_font_info["weight"] or "", re.I):
```

Comment:
This is the `FcIsWeight` in fontconfig, which calls `FcStringIsConst`, which calls `FcStrCmpIgnoreBlanksAndCase`, which ignores _any_ spaces. Do we actually normalize `ps_font_info`, or does that replacement need to be done here too?

Reply:
I probably just missed that, fixed accordingly.